airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py

```diff
@@ -0,0 +1,64 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Add map_index to Log.
+
+Revision ID: 75d5ed6c2b43
+Revises: 4eaab2fe6582
+Create Date: 2022-03-15 16:35:54.816863
+"""
+from alembic import op
+from sqlalchemy import Column, Integer
+
+# Revision identifiers, used by Alembic.
+revision = '75d5ed6c2b43'
+down_revision = '4eaab2fe6582'
+branch_labels = None
+depends_on = None
+airflow_version = '2.3.0'
+
+
+def upgrade():
+    """Add map_index to Log."""
+    op.add_column("log", Column("map_index", Integer, server_default="-1"))
+    # Only set map_index if the log is related to a task instance.
+    op.execute(
+        """
+        update log set map_index = -1
+        where dag_id is not null and task_id is not null and execution_date is not null
+        """
```

Comment:
I think that the `server_default` above will set the `map_index` regardless of whether dag_id etc is null